Wox/Helper/SingleInstance.cs

```diff
@@ -338,56 +317,43 @@ private static IList<string> GetCommandLineArgs( string uniqueApplicationName )
         }
 
         /// <summary>
-        /// Creates a remote service for communication.
+        /// Creates a remote server pipe for communication. 
+        /// Once receives signal from client, will activate first instance.
         /// </summary>
         /// <param name="channelName">Application's IPC channel name.</param>
-        private static void CreateRemoteService(string channelName)
+        private static async void CreateRemoteService(string channelName)
```

Comment:
this should be `async Task`

Reply:
Also InitializeFirstInstance should be async and CreateRemoteService and SingleFirstInstance should be awaited.  aync should be used all the way up the call chain, and async void should be avoided.

Reply:
If I use async all the way up the call chain I will end up wit the Main being async, which is not possible. Dont think that the calls need to be awaited on `InitializeFirstInstance `, they can keep running while the rest of the code proceeds.

Reply:
This can be problematic if the application ever needs to handle exceptions thrown in these methods.  It is possible to go all the way up the chain.  C# 7.1 supports async main, I haven't tried this in a wpf application, though.   Alternatively you can create your own MainAysnc method and use Task.GetAwaiter().GetResult().  Both methods are outlined here: 

https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7-1#async-main

Reply:
Another possibility is to implement async voids that don't raise exceptions. This can be done  by implementing a task extension that returns an async void and awaits for the task, capturing any exceptions that are raised.

e.g

```
    public static async void AwaitAndHandle(this Task task, Func<Task> handleException = null)
        {
            try
            {
                await task;
            }
            catch (Exception)
            {
                if (handleException != null)
                {
                    await handleException();
                }
            }
        }
```

In your logic, you can convert `private static async void CreateRemoteService(string channelName)` to `private static async Task CreateRemoteServiceAsync(string channelName)` and simply call `CreateRemoteServiceAsync(string channelName).AwaitAndHandle();` wherever required.